Issue31 client

ide-an opened the pull request:
#31 に関する実装。

このpull requestで実装したことは以下のとおり
- GETおよびPOSTによる通信の実装
- postQuestion()の実装
- getQuestions()の実装

これらに関連したテストも記述した。

ide-an committed:
GETによる通信の実装

TODO:リクエスト時の パラメータの受け渡し
TODO:POSTの実装
TODO:Cookieまわりの動作確認

ide-an committed:
GETによる通信の実装

TODO:リクエスト時の パラメータの受け渡し
TODO:POSTの実装

ide-an committed:
format code

ide-an committed:
改行文字の変更

ide-an committed:
cookieまわりのデバッグ表示

cookieの受け渡しについて正常動作することを確認した。

ide-an committed:
改行コードを修正

ide-an committed:
リクエスト先変更

ide-an committed:
改行コードの変更

ide-an committed:
Fix: パス中の'/'がエンコードされるバグを修正

ide-an committed:
merged issue31-client

ide-an committed:
クエリパラメータまわり実装

ide-an committed:
EntityUtilsを使用

ide-an committed:
parse JSON

ide-an committed:
refactor sendByGet

ide-an committed:
implement communication by POST

ide-an committed:
implement getQuestions() and postQuestion()

ide-an committed:
getQuestions()についてテストを記述
現状ではJSONまわりのテストがうまくいかない
どうやらrobolectricの問題？
  https://groups.google.com/group/robolectric/browse_thread/thread/4d2c4ca6eba5fa86

draftcode commented:
AndroidのHTTPClientあたりのライブラリは、ApacheのHttpClientと同様なものがが用いられています。ApacheのHttpClientにはチュートリアルがあるので、それを参照してみると良い実装やテストが書けると考えられます。

http://hc.apache.org/httpcomponents-client-ga/tutorial/html/

draftcode committed:
Fix Robolectric's JSON problem.

Robolectric depends on JSON library but it doesn't include that library. It
seems that the developers of Robolectric acknowledged this issue and tried to
fix it. But in fact it doesn't fixed.

For the time being the issue not fixed, we add the JSON library for
Robolectric.

References:

* [Issue #4: Failing test - pivotal/robolectric](https://github.com/pivotal/robolectric/issues/4)
* [JSONObject failing with Stub! error on 1.0-RC1 - Google Groups](https://groups.google.com/d/topic/robolectric/TSxMpuul-oY/discussion)

draftcode committed:
Inject API endpoint's URL.

draftcode commented:
JSONのライブラリの問題、ありがとうございます。コミットメッセージでも書きましたが、Robolectricの開発者の方も認識はしていたようで、0.9.8あたりでは直しているんですが、再発しているようですね。とりあえず、こちら側でJSONライブラリを添付することで対処しました。

また、HQTPのAPIエンドポイントがローカルアドレスになっていて、テストが通らなかったので、その文字列を注入するようにしました。

JUnitのテストが通ることと、実機で動かしてみて動くことを確認しました。(実機で動かした時にhqtp.orgにアクセスするので質問が取得できず、質問がありませんと表示されてちょっとビビリました。)
